File: JUnit/saucery-core/src/main/java/au/com/fullcirclesolutions/saucery/utils/SauceryConstants.java
package au.com.fullcirclesolutions.saucery.utils;

public class SauceryConstants {
    public static final String FORWARDSLASH = "/";
    public static final String EMPTY_STRING = "";

    //ACTIVATION DETAILS
    public static final String MY_USERNAME = "fullcicle";
    public static final String COMPANYNAME = "fullcirclesolutions";
    public static final String PRODUCTNAME = "Saucery";
    public static final String VERSION = "4.0.0.0";
    public static final String EDITION = "enterprise";
    public static final String KEYNAME = "saucerykey";
    public static final String SAUCERY_LICENCE_FILE = "\\Saucery4.lic"; //need the backslash here!
    public static final String PUBLIC_KEY = "BwIAAACkAABSU0EyAAQAAAEAAQDtZrs0gFK/FqIRSFYO8wqbwWUqk0Vul1ueiuOTdQrdkRy4bPT0Z8FTOWpAwlJXyxeJ+WgQmJlITEcb7VH0cICpJeWYPCGACUkqibDUGvNS/shLg4OoxPxP7u8kDnomqCSxhH5W0HXVrY73U3mq9tTlVVLvGqTbBG2/JMwnRSk70jURWrm1Py/vsMnluJU5kGUcbgFwF/ALCYBEttrmhavdgixswTfPT3LUg8IK44vKpVkSqegav+LpZlxwVvHLZvjZrfo8O3ZccC+XdFp+CdJdDbJ8KC8amqWz5hgeSciHzq2JRKppNhOslQQ/wDoxUJP7HDST4zgFf1UzGWHMdKnYAdoadFfCQ8kmHlmxEXWkVud9sZkG6fJbouTH2lCG0Ad5Z2URkgZJHhUTanbHqzzi8uS7AEM7DuIEwO99IpACEHlr8Il+Rbz7xIq3b/u8V7bhORoua1jmX5Wg6Du8zdpziWUxR+Yg53/A0BkWcgBci24Tun1PoDKuzK8yHPZiSEVYmX/bw6Htbzi+XyFNGF1TMfJaohhHi421MSzZMz+qKwSPQ+Ps0QJxPfvi/SDHUtlnkhmKEQ0gq5uu2cFIgjKUIb+S1NYJF3yW4oIiVBfrrBXx2wXkbRgDHlTPyrOXuFhzpjgMGjonEYMfHUSqFshObg6Go3r50jJQQDn7b230fhdtanz9M24raQRTlrC5Q6eEN0TqrwiTQoxpxa75mUHfd7YpHnButRC8YjrjdniHaxYSdk54TV3FPyN9niZfHzo=";
    
    //TEST URL
    public static final String SAUCELABS_HUB = "http://%s:%s@ondemand.saucelabs.com:80/wd/hub";

    //CAPABILITIES
    //public static final String RESOLUTION_CAPABILITY = "resolution";
    //public static final String SCREENRESOLUTION_CAPABILITY = "screen-resolution";
    public static final String SAUCE_APPIUM_VERSION_CAPABILITY = "appium-version";
    public static final String SAUCE_APPIUM_VERSION = "1.4.7";
    public static final double APPIUM_IOS_MINIMUM_VERSION = 6.1;
    public static final double APPIUM_ANDROID_MINIMUM_VERSION = 4.4;
    public static final String SAUCE_SESSIONNAME_CAPABILITY = "name";
    public static final String SAUCE_BUILDNAME_CAPABILITY = "build";
    public static final String SAUCE_USERNAME_CAPABILITY = "username";
    public static final String SAUCE_ACCESSKEY_CAPABILITY = "accessKey";
    public static final String SAUCE_BROWSER_NAME_CAPABILITY = "browserName";
    public static final String SAUCE_PLATFORM_CAPABILITY = "platform";
    public static final String SAUCE_PLATFORM_NAME_CAPABILITY = "platformName";
    public static final String SAUCE_PLATFORM_VERSION_CAPABILITY = "platformVersion";
    public static final String SAUCE_VERSION_CAPABILITY = "version";
    public static final String SAUCE_DEVICE_CAPABILITY = "device";
    public static final String SAUCE_DEVICE_NAME_CAPABILITY = "deviceName";
    public static final String SAUCE_DEVICE_ORIENTATION_CAPABILITY = "device-orientation";
    public static final String SAUCE_NATIVE_APP_CAPABILITY = "app";
    public static final String SAUCE_VUOP_CAPABILITY = "video-upload-on-pass";

    //SAUCE ENVIRONMENT VARIABLES
    public static final String SAUCE_USER_NAME = "SAUCE_USER_NAME";
    public static final String SAUCE_API_KEY = "SAUCE_API_KEY";
    public static final String SAUCE_ONDEMAND_BROWSERS = "SAUCE_ONDEMAND_BROWSERS";
    public static final String SAUCE_NATIVE_APP = "SAUCE_NATIVE_APP";
    public static final String SAUCE_USE_CHROME_ON_ANDROID = "SAUCE_USE_CHROME_ON_ANDROID";
    public static final String JENKINS_BUILD_NUMBER = "JENKINS_BUILD_NUMBER";
    public static final String BAMBOO_BUILD_NUMBER = "SAUCE_BAMBOO_BUILDNUMBER";

    //TUNNELLING
    public static final int TUNNEL_CONNECT_RETRY_COUNT = 3;

    //FLOW CONTROL
    public static final int SAUCELABS_FLOW_WAIT = 1000;

    //DEVICES
    public static final String LINUX = "Linux";
    public static final String APPLE_IOS = "Apple IOS";
    public static final String IOS_PLATFORM = "iOS";
    public static final String ANDROID_PLATFORM = "ANDROID";
    public static final String ANDROID_DEVICE = "android";
    public static final String GOOGLE = "Google";
    public static final String LG = "LG";
    public static final String SAMSUNG = "Samsung";
    public static final String APPLE_IPHONE = "iphone";
    public static final String APPLE_IPAD = "ipad";
    public static final String ANDROID = "android";
    public static final String SAFARI_BROWSER = "Safari";
    public static final String CHROME_BROWSER = "chrome";
    public static final String DEFAULT_ANDROID_BROWSER = "browser";
    public static final String IPAD_SIMULATOR = "iPad Simulator";
    public static final String IPHONE_SIMULATOR = "iPhone Simulator";
    public static final String IPAD_DEVICE = "iPad";
    public static final String IPHONE_DEVICE = "iPhone";

    //MISCELLANEOUS
    public static final String APPLE_TITLE = "Apple";
    public static final String PASSED = "passed";
    public static final String FAILED = "failed";
    public static final String NULL_STRING = "null";
    public static final String SPACE = " ";
    public static final String TWO_SPACES = "  ";
    public static final String LEFT_BRACKET = "(";
    public static final String LEFT_SQUARE_BRACKET = "[";
    public static final String COLON = ":";
    public static final String UNDERSCORE = "_";
    public static final String RESOLUTION = "1280x1024";
    public static final String YEAR_FORMAT = "yyyy";
    
    //NOTIFICATION MESSAGES
    public static final String CONSOLE_RUNNER = "%s Console Runner";
    public static final String COPYRIGHT_NOTICE = "Copyright (C) 2008-%s Andrew Gray";
    public static final String TESTING_ON = "Testing on %s %s.";
    public static final String NO_PLATFORMS = "Platforms is null";
    public static final String ON_DEMAND = "OnDemand=%s";
    public static final String LICENCED_VERSION = "You are using the licenced version of %s.";
    public static final String TRIAL_VERSION = "You are using the trial version of %s.";
    public static final String DAYS_REMAINING = "You have %s days remaining.";

    //SETTING UP
    public static final String SETTING_UP = "Setting up %s for %s";
    public static final String ANDROID_ON_APPIUM = "Android on Appium";
    public static final String IOS_ON_APPIUM = "IOS on Appium";
    public static final String ANDROID_ON_WEBDRIVER = "Android on WebDriver";
    public static final String IOS_ON_WEBDRIVER = "IOS on WebDriver";
    public static final String DESKTOP_ON_WEBDRIVER = "Desktop on WebDriver";

    //TEST VISIBILITY
    public static final String VISIBILITY_KEY = "public";
    public static final String VISIBILITY_PUBLIC = "public";
    public static final String VISIBILITY_PUBLIC_RESTRICTED = "public restricted";
    public static final String VISIBILITY_SHARE = "share";
    public static final String VISIBILITY_TEAM = "team";
    public static final String VISIBILITY_PRIVATE = "private";
}
/*
 * Copyright: Andrew Gray, Full Circle Solutions
 * Date: 15th November 2013
 */

File: Saucery/Util/SauceryConstants.cs
namespace Saucery.Util {
    public class SauceryConstants {
        //ACTIVATION DETAILS
        internal const string MY_USERNAME = "saucefauge";
        public static readonly string MY_USERNAME_LOWER = MY_USERNAME.ToLower();
        internal const string COMPANYNAME = "SauceForge";
        internal const string EDITION = "Enterprise";
        internal const string KEYNAME = "SauceryKey";
        internal const string REGISTRYROOT = @"SOFTWARE\" + COMPANYNAME + BACKSLASH;
        internal const string LICENCE_DATE_FORMAT = "ddMMyyyy";
        //internal const string PUBLIC_KEY = "44C9B9B2D6C2D11CBF7CD9DF9C37178462A1CE514FCED53CE9B04561C35614C5E525AEE5E3064CFB68D676522F92992721A60374032A200D07CE956B25B87B81";
        //internal const string FIRST_KEY = "0CDCF65343FEDEE34702856B251025AD80D4FE58C2481B584D7FB246BAC23A6DB4E44E5654A856F182A673D6953FD42725FAA316DE6A78059DECE599BF233D00";
        internal const int FREE_TRIAL_PERIOD = 30;
        internal const int PAID_LICENCE_PERIOD = 366;

        internal const string PRODUCTNAME = "Saucery";
        internal const string VERSION = "4.0.0.0";
        internal const string SAUCERY_LICENCE_FILE = @"\Saucery.lic"; //need the backslash here!
        internal const string LICENCE_PRODUCTCODE = "N3";
        internal const string TRIAL_PRODUCTCODE = "3T";
        internal static readonly int[] NUNIT3POSARRAY = { 3, 6, 9, 12, 15, 18, 21, 24, 27, 30 };
        internal const string SAUCERY_ASSEMBLY = "<Reference Include=\"Saucery";
        internal const string SAUCERYTESTER_ASSEMBLY = "<ProjectReference Include=\"..\\Saucery\\Saucery.csproj\">";

        internal const string SOLUTION_MASK = ASTERIX + ".sln";
        internal const string PROJECT_EXTENSION = ".csproj";
        internal const string PROJECT_MASK = ASTERIX + PROJECT_EXTENSION;
        internal const string ASSEMBLYINFO_RELATIVE_PATH = @"\Properties\AssemblyInfo.cs";
        internal const string ASSEMBLY_FILE_GUID_LINE = "[assembly: Guid";
        internal const string SOLUTION_FILE_PROJECT_LINE = "Project(";

        //TEST URL
        internal const string SAUCELABS_HUB = "http://ondemand.saucelabs.com:80/wd/hub";

        //CAPABILITIES
        //internal const string RESOLUTION_CAPABILITY = "resolution";
        //internal const string SCREENRESOLUTION_CAPABILITY = "screen-resolution";
        public const string LATEST_APPIUM_VERSION = "1.22.0";
        internal const string SAUCE_APPIUM_VERSION_CAPABILITY = "appiumVersion";
        internal const string SELENIUM_VERSION_CAPABILITY = "seleniumVersion";
        internal const string LATEST_SELENIUM_VERSION = "3.141.0";  //"latest" does not work.
        internal const double APPIUM_IOS_MINIMUM_VERSION = 6.1;
        internal const double APPIUM_ANDROID_MINIMUM_VERSION = 4.4;
        internal const string SAUCE_SESSIONNAME_CAPABILITY = "name";
        internal const string SAUCE_BUILDNAME_CAPABILITY = "build";
        internal const string SAUCE_USERNAME_CAPABILITY = "username";
        internal const string SAUCE_ACCESSKEY_CAPABILITY = "accessKey";
        internal const string SAUCE_BROWSER_NAME_CAPABILITY = "browserName";
        internal const string SAUCE_PLATFORM_CAPABILITY = "platform";
        internal const string SAUCE_PLATFORM_NAME_CAPABILITY = "platformName";
        //internal const string SAUCE_PLATFORM_VERSION_CAPABILITY = "platformVersion";
        internal const string SAUCE_PLATFORM_VERSION_CAPABILITY = "PlatformVersion";
        internal const string SAUCE_APPIUM_PLATFORM_VERSION_CAPABILITY = "platformVersion";
        internal const string SAUCE_VERSION_CAPABILITY = "version";
        internal const string SAUCE_DEVICE_CAPABILITY = "device";
        //internal const string SAUCE_DEVICE_NAME_CAPABILITY = "deviceName";
        internal const string SAUCE_DEVICE_NAME_CAPABILITY = "DeviceName";
        internal const string SAUCE_APPIUM_DEVICE_NAME_CAPABILITY = "deviceName";
        internal const string SAUCE_DEVICE_ORIENTATION_CAPABILITY = "deviceOrientation";
        internal const string SAUCE_NATIVE_APP_CAPABILITY = "app";
        internal const string SAUCE_VUOP_CAPABILITY = "videoUploadOnPass";
        internal const string SAUCE_OPTIONS_CAPABILITY = "sauce:options";

        //SAUCE ENVIRONMENT VARIABLES
        public const string SAUCE_USER_NAME = "SAUCE_USER_NAME";
        public const string SAUCE_API_KEY = "SAUCE_API_KEY";
        internal const string SAUCE_ONDEMAND_BROWSERS = "SAUCE_ONDEMAND_BROWSERS";
        internal const string SAUCE_NATIVE_APP = "SAUCE_NATIVE_APP";
        internal const string SAUCE_USE_CHROME_ON_ANDROID = "SAUCE_USE_CHROME_ON_ANDROID";
        internal const string JENKINS_BUILD_NUMBER = "JENKINS_BUILD_NUMBER";
        internal const string BAMBOO_BUILD_NUMBER = "SAUCE_BAMBOO_BUILDNUMBER";
        internal const string BUILD_NUMBER = "BUILD_BUILDNUMBER";
        internal const string RECOMMENDED_APPIUM_VERSION = "RECOMMENDED_APPIUM_VERSION";

        //TUNNELLING
        internal const int TUNNEL_CONNECT_RETRY_COUNT = 3;

        //REST
        internal const string JSON_CONTENT_TYPE = "application/json";
        internal const string RESTAPI_LIMIT_EXCEEDED_INDICATOR = "API rate limit exceeded for";
        internal const string RESTAPI_LIMIT_EXCEEDED_MSG = "API rate limit exceeded. We have to wait. See https://saucelabs.com/blog/announcing-new-rest-api-rate-limits for more details.";
        internal const string SAUCE_REST_BASE = "https://api.us-west-1.saucelabs.com/rest/v1/";
        internal const string ACCOUNT_CONCURRENCY_REQUEST = "users/{0}/concurrency";
        internal const string JOB_REQUEST = "{0}/jobs/{1}";
        internal const string RECOMMENDED_APPIUM_REQUEST = "info/platforms/appium";
        internal const string SUPPORTED_PLATFORMS_REQUEST = "info/platforms/all";
        internal const string JSON_SEGMENT_CONTAINER = "{{{0}}}";
        internal const string NUGET_API = "https://packages.nuget.org/api/v2";
        internal const string NUGET_VERSION = "{0}.{1}.{2}";

        //FLOW CONTROL
        internal const int SAUCELABS_FLOW_WAIT = 3000;

        //DEVICES
        internal const string LINUX = "Linux";
        internal const string APPLE_IOS = "Apple IOS";
        internal const string ANDROID_PLATFORM = "ANDROID";
        internal const string ANDROID_DEVICE = "android";
        internal const string GOOGLE = "Google";
        internal const string LG = "LG";
        internal const string SAMSUNG = "Samsung";
        internal const string APPLE_IPHONE = "iphone";
        internal const string APPLE_IPAD = "ipad";
        internal const string ANDROID = "Android";
        internal const string SAFARI_BROWSER = "Safari";
        internal const string CHROME_BROWSER = "Chrome";
        internal const string IPAD_SIMULATOR = "iPad Simulator";
        internal const string IPHONE_SIMULATOR = "iPhone Simulator";
        internal const string IPAD_DEVICE = "iPad";
        internal const string IPHONE_DEVICE = "iPhone";

        //MISCELLANEOUS
        internal const string APPLE_TITLE = "Apple";
        internal const string PASSED = "passed";
        internal const string FAILED = "failed";
        internal const string NULL_STRING = "null";
        public const string DOT = ".";
        internal const string SPACE = " ";
        internal const string TWO_SPACES = "  ";
        internal const string ASTERIX = "*";
        internal const char SPACE_CHAR = ' ';
        internal const string BACKSLASH = @"\";
        internal const string LEFT_BRACKET = "(";
        internal const char LEFT_BRACKET_CHAR = '(';
        internal const char RIGHT_BRACKET_CHAR = ')';
        internal const string LEFT_SQUARE_BRACKET = "[";
        internal const string COLON = ":";
        internal const string UNDERSCORE = "_";
        internal const string RESOLUTION = "1280x1024";
        internal const string YEAR_FORMAT = "yyyy";

        //NOTIFICATION MESSAGES
        //internal const string CONSOLE_RUNNER = "{0} Console Runner {1}";
        internal const string CONSOLE_RUNNER = "{0} Console Runner";
        internal const string COPYRIGHT_NOTICE = "Copyright (C) 2008-{0} Andrew Gray";
        internal const string TESTING_ON = "Testing on {0} {1}";
        internal const string NO_PLATFORMS = "Platforms is null";
        internal const string ON_DEMAND = "OnDemand={0}";
        internal const string LICENCED_VERSION = "You are using the licenced version of {0}.";
        internal const string TRIAL_VERSION = "You are using the trial version of {0}.";
        internal const string DAYS_REMAINING = "You have {0} days remaining.";
        internal const string NEW_VERSION_AVAILABLE = "You are using version {0}. Version {1} is available!";

        //SETTING UP
        internal const string SETTING_UP = "Setting up {0} for {1}";
        internal const string SETTING_UP_APPIUM = "Setting up {0} for {1} {2}";
        internal const string ANDROID_ON_APPIUM = "Android on Appium";
        internal const string IOS_ON_APPIUM = "IOS on Appium";
        internal const string ANDROID_ON_WEBDRIVER = "Android on WebDriver";
        internal const string IOS_ON_WEBDRIVER = "IOS on WebDriver";
        internal const string DESKTOP_ON_WEBDRIVER = "Desktop on WebDriver";

        //TEST VISIBILITY
        internal const string VISIBILITY_KEY = "public";
        internal const string VISIBILITY_PUBLIC = "public";
        internal const string VISIBILITY_PUBLIC_RESTRICTED = "public restricted";
        internal const string VISIBILITY_SHARE = "share";
        internal const string VISIBILITY_TEAM = "team";
        internal const string VISIBILITY_PRIVATE = "private";

        //OPTIONS BROWSER SUPPORT
        internal const int MIN_SAFARI_SUPPORTED_VERSION = 8;
        internal const int MIN_FIREFOX_SUPPORTED_VERSION = 78;
        internal const int MIN_CHROME_SUPPORTED_VERSION = 75;
        internal const int MIN_EDGE_SUPPORTED_VERSION = 79;
        internal const int MIN_IE_SUPPORTED_VERSION = 9;

        internal const int MAX_SAFARI_SUPPORTED_VERSION = 15;
        internal const int MAX_FIREFOX_SUPPORTED_VERSION = 98;
        internal const int MAX_CHROME_SUPPORTED_VERSION = 99;
        internal const int MAX_EDGE_SUPPORTED_VERSION = 99;
        internal const int MAX_IE_SUPPORTED_VERSION = 11;

        internal const string NOT_SUPPORTED_MESSAGE = "Browser version not supported by DriverOptions. Please see: https://wiki.saucelabs.com/display/DOCS/2018/03/27/Introducing+Selenium+WebDriver+W3C+Protocol+Beta.";

        //PLATFORMS
        public const string PLATFORM_WINDOWS_11 = "Windows 11";
        public const string PLATFORM_WINDOWS_10 = "Windows 10";
        public const string PLATFORM_WINDOWS_81 = "Windows 2012 R2";
        public const string PLATFORM_WINDOWS_8 = "Windows 2012";
        public const string PLATFORM_WINDOWS_7 = "Windows 2008";
        public const string PLATFORM_MAC_12 = "Mac 12";
        public const string PLATFORM_MAC_11 = "Mac 11";
        public const string PLATFORM_MAC_1015 = "Mac 10.15";
        public const string PLATFORM_MAC_1014 = "Mac 10.14";
        public const string PLATFORM_MAC_1013 = "Mac 10.13";
        public const string PLATFORM_MAC_1012 = "Mac 10.12";
        public const string PLATFORM_MAC_1011 = "Mac 10.11";
        public const string PLATFORM_MAC_1010 = "Mac 10.10";
        public const string PLATFORM_IOS = "iOS";

        //BROWSERS
        public const string BROWSER_CHROME = "chrome";
        public const string BROWSER_FIREFOX = "firefox";
        public const string BROWSER_IE = "internet explorer";
        public const string BROWSER_EDGE = "MicrosoftEdge";
        public const string BROWSER_SAFARI = "safari";

        //SCREEN RESOLUTIONS
        public const string SCREENRES_800_600 = "800x600";
        public const string SCREENRES_1024_768 = "1024x768";
        public const string SCREENRES_1152_720 = "1152x720";
        public const string SCREENRES_1152_864 = "1152x864";
        public const string SCREENRES_1152_900 = "1152x900";
        public const string SCREENRES_1280_720 = "1280x720";
        public const string SCREENRES_1280_768 = "1280x768";
        public const string SCREENRES_1280_800 = "1280x800";
        public const string SCREENRES_1280_960 = "1280x960";
        public const string SCREENRES_1280_1024 = "1280x1024";
        public const string SCREENRES_1376_1032 = "1376x1032";
        public const string SCREENRES_1400_1050 = "1400x1050";
        public const string SCREENRES_1440_900 = "1440x900";
        public const string SCREENRES_1600_900 = "1600x900";
        public const string SCREENRES_1600_1200 = "1600x1200";
        public const string SCREENRES_1680_1050 = "1680x1050";
        public const string SCREENRES_1920_1080 = "1920x1080";
        public const string SCREENRES_1920_1200 = "1920x1200";
        public const string SCREENRES_1920_1440 = "1920x1440";
        public const string SCREENRES_2048_1152 = "2048x1152";
        public const string SCREENRES_2048_1536 = "2048x1536";
        public const string SCREENRES_2360_1770 = "2360x1770";
        public const string SCREENRES_2560_1600 = "2560x1600";

    }
}
/*
 * Copyright Andrew Gray, SauceForge
 * Date: 12th July 2014
 * 
 */